Adding example of Rule 3 - Markdown and Creating an Index

christhorne opened the pull request:
Hello!

I worked on a Jupyter Notebook example which I believe highlights Rule 3. Specifically, I wanted to give an example of a notebook which used markdown throughout, and, also had an index in the beginning which provided a quick way to jump to the respective sections.

Also, this is my first Github fork and pull request. So, ** please ** let me know if I am doing this correctly as I am very excited to contribute. I'm doing this for a Master's program I am a part of in Applied Data Science (https://www.si.umich.edu/programs/master-applied-data-science-online), so I appreciate any feedback as well on if you think the Jupyter notebook helps demonstrate the rule.

Thank you! - Chris

christhorne committed:
Add files via upload

Jupyter Notebook Example of Rule 3

christhorne committed:
Update README.md

christhorne committed:
Update README.md

pwrose reviewed (changes requested):
Hi Chris,
Congrats on your first pull request!

I have a few suggestions:
1. Create a top-level folder, e.g., notebooks (then this folder can be used for other example notebooks)
2. Move your notebook to the notebooks folder
3. Simplify the name of the notebook and avoid non-alphanumeric characters such as "&". Thinking about the future, you may want to call the notebook: Rule3_Index_Markdown. Then somebody else may add Rule2_xxx, and so on. That keeps the repository organized.
4. In the Readme file, use a generic title, e.g., Example Notebooks, and link it to the notebooks folder.
5. Generally, avoid using ":" at the end of Jupyter Notebook headers
6. In the Helpful References section, avoid showing long URLs, use links instead.

Nice job!

Thanks again for your first contribution.

christhorne committed:
Create README.md

christhorne committed:
Delete README.md

christhorne committed:
Adding notebook and new Jupyter Notebook file

christhorne committed:
Update README.md

christhorne commented:
Thank you for the feedback!

1. Done.
2. Done.
3. Good to know, thank you! Done.
4. Done.
5. Fixed.
6. Fixed.

Happy to contribute.

Thanks - Chris
